src/_pytest/python.py

```diff
@@ -902,6 +902,25 @@ def make_unique_parameterset_ids(self) -> list[str | _HiddenParam]:
         resolved_ids = list(self._resolve_ids())
         # All IDs must be unique!
         if len(resolved_ids) != len(set(resolved_ids)):
+            if self.config and self.config.cache._config.getoption(
+                "require_unique_paramset_ids"
+            ):
+                duplicate_indexs = defaultdict(list)
+                for i, val in enumerate(resolved_ids):
+                    duplicate_indexs[val].append(i)
+
+                # Keep only duplicates
+                duplicates = {k: v for k, v in duplicate_indexs.items() if len(v) > 1}
+                raise Exception(f"""
```

Comment:
`Exception` will show this message as an ugly internal error, with a long traceback showing pytest's internals. I think raising `UsageError` will provide a better message.